xtask/src/collect_endpoints/helix.rs

```diff
@@ -208,3 +217,96 @@ impl<'a> HelixRow<'a> {
         ))
     }
 }
+
+
+pub fn distance(src: &str, tar: &str) -> usize {
+    let ignore_phrases = vec!["in_channel", "by_id", "from_ids", "from_id", "from_logins", "from_login"];
+
+    fn remove_ignored_phrases(s: &str, ignore_phrases: &[&str]) -> String {
+        let mut result = s.to_string();
+        for phrase in ignore_phrases {
+            result = result.replace(phrase, "_");
+        }
+        result
+    }
+
+    fn depluralize(token: &str) -> &str {
+        if token.ends_with('s') && token.len() > 1 {
+            &token[..token.len() - 1]
+        } else {
+            token
+        }
+    }
+
+    let src_cleaned = remove_ignored_phrases(src, &ignore_phrases);
+    let tar_cleaned = remove_ignored_phrases(tar, &ignore_phrases);
+
+    let src_tokens: Vec<&str> = src_cleaned
+        .split('_')
+        .filter(|token| !token.is_empty())
+        .map(depluralize)
+        .collect();
+
+    let tar_tokens: Vec<&str> = tar_cleaned
+        .split('_')
+        .filter(|token| !token.is_empty())
+        .map(depluralize)
+        .collect();
+
+    let min_len = usize::min(src_tokens.len(), tar_tokens.len());
+    let mut distance = 0;
+
+    // Compare tokens like `editor` and `emote`
+    for i in 0..min_len {
+        // XXX: could make early tokens penalize heavier
+        let weight = 2;
+
+        if src_tokens[i] == tar_tokens[i] {
+            continue;
+        } else {
+            let token_distance = super::levenshtein(src_tokens[i], tar_tokens[i]);
+            distance += weight * token_distance;
+        }
+    }
+
+    // penalize extra tokens
+    // for example, `get_channel_editors`, `get_channels_from_ids` == 24
+    if src_tokens.len() > min_len {
+        for token in &src_tokens[min_len..] {
+            let weight = 4; // higher penalty for extra tokens
```

Comment:
this weight may be too high, should maybe be 3, a trailing single character after stemming/trimming should probably not result in a distance of 4

Reply:
My NLP skills are too bad, but it does give the correct output, no?

Reply:
yeah, it does :)